enterprise/reporting/setup_test.go

```diff
@@ -77,8 +82,8 @@ func TestSetupForNoop(t *testing.T) {
 					EnterpriseToken: "dummy-token",
 				}
 			}
-			med := NewReportingMediator(logger.NOP, f.EnterpriseToken)
-			require.Equal(t, med.createErrorReportInstance(&backendconfig.NOOP{}), &NOOP{})
+			med := NewReportingMediator(context.Background(), logger.NOP, f.EnterpriseToken, &backendconfig.NOOP{})
+			require.Len(t, med.delegates, tc.expectedDelegates)
```

Comment:
As per the existing code, should we not create any reporting instance here? Since we are comparing it with NOOP?

Reply:
please elaborate, we are comparing expected number of delegates, right?

Reply:
It's clarified in [here](https://github.com/rudderlabs/rudder-server/pull/3912#discussion_r1338104614). Marking it as resolved.